src/services/Notion.ts

```diff
@@ -0,0 +1,61 @@
+import { Client } from '@notionhq/client';
+import { Service } from './Service';
+import { NotionConfig } from './Config';
+
+/** Class representing an interaction with Notion. */
+export class Notion implements Service<NotionConfig> {
+  public static instance: Notion;
+  public notion: Client | null = null;
+  public configuration?: NotionConfig;
+  /**
+   * Create and returns a Notion instance.
+   *
+   * @param configuration - Configuration for connection
+   * @returns {Notion}
+   */
+  constructor(configuration: NotionConfig) {
+    if (!Notion.instance) {
```

Comment:
Looks strange. Is there any reason to not to use regular class? It's a simple service.

Reply:
We talked about singleton https://github.com/codex-team/codex-surveys/pull/2#discussion_r1039471437
And about implementation, if it need i can change it https://github.com/codex-team/codex-surveys/pull/2#discussion_r1042493124

Reply:
Looks like there is a problem with the module architect. Singleton is a marker for that.

- `sendDataToNotion` shouldn't be util. 
- app should have an entry point
- app should instantiate a 'sender' provider (Notion)
- other modules should not have an access to the sender instance.
- app the sender provider should have a `send` method
- app will create an initial ui
- app will create a Form
- app will have an access to the Form's data
- app will have an access to the 'sender' provider
- app will send data using a sender

It's quite simple. No singleton is required for that. 

I think you should refactor the entry point. For now, it's just a `createWidget` function, but there should be an App itself.